atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/KeyValueService.java

```diff
@@ -426,6 +426,17 @@ void deleteAllTimestamps(TableReference tableRef,
             TableReference tableRef,
             CandidateCellForSweepingRequest request);
 
+    /**
+     * Scans a range of a table and returns the row keys in that range.
+     * <p>
+     * This endpoint specifically avoids paging through each full row.
+     * <p>
+     * The batch size limits the number of results from each request.
+     */
+    default List<byte[]> getRowKeysInRange(TableReference tableRef, byte[] startRowInclusive, int batchSize) {
```

Comment:
How do we feel about not adding more methods that use `byte[]`? It's kind of awkward to mix and match `ByteString` (or whatever alternative) and `byte[]` in the same interface, but not convinced that perpetuating the use of byte arrays in newly added methods is great either

Reply:
it might be possible we could probably generate overloads to keep backcompat or something, where the base interface will be in terms of `ByteString`s and then the public facing type: `KeyValueService` will keep the `byte[]` stuff around. 

On the other hand, are there really that many places that will use a `byte[]` directly? We should be able to make most of these changes.

> N.B. this is for api surfaces, internals can change at a later cadence I suppose.